Sources/Flows/OAuth2.swift

```diff
@@ -489,7 +489,57 @@ open class OAuth2: OAuth2Base {
 			callback(nil, error.asOAuth2Error)
 		}
 	}
-	
+    
+    // MARK: - Exchange Access Token For Resource
+    open func tokenRequestForExchangeAccessTokenForResource(resourceId: String, resourceType: String, params: OAuth2StringDict? = nil) throws -> OAuth2AuthRequest {
+        guard let accessToken = clientConfig.accessToken, !accessToken.isEmpty else {
+            throw OAuth2Error.noAccessToken
+        }
+        guard let resourceUrl = clientConfig.resourceURL, !resourceUrl.absoluteString.isEmpty else {
+            throw OAuth2Error.noResourceURL
+        }
+        let req = OAuth2AuthRequest(url: (clientConfig.tokenURL ?? clientConfig.authorizeURL))
+        req.params["grant_type"] = "urn:ietf:params:oauth:grant-type:token-exchange"
+        req.params["client_id"] = clientConfig.clientId
+        req.params["resource"] = resourceUrl.absoluteString + "/" + resourceType + "/" + resourceId
```

Comment:
It is always safer to use methods provided by URL struct to manipulate URLs:
```suggestion
        		req.params["resource"] = resourceUrl
				.appendingPathComponent(resourceType)
				.appendingPathComponent(resourceId)
				.absoluteString
```

Reply:
Reflecting back on this change, I think we should avoid the structure `/{resourceType}/{resourceId}`, as in OAuth spec the resource can be in any format.